src/drivers/ports/esp32/can_port.i

```diff
@@ -139,33 +216,53 @@ static void isr(void *arg_p)
 
     regs_p = self_p->dev_p->regs_p;
 
-    /* Read the interrupt status. */
+    /* Read the interrupt status and status registers. */
     interrupt = regs_p->INT;
 
     /* Handle TX complete interrupt. */
-    if ((interrupt & ESP32_CAN_INT_TX) != 0) {
+    if (interrupt & ESP32_CAN_INT_TX) {
         if (self_p->txsize > 0) {
             write_frame_to_hw(regs_p, self_p->txframe_p);
             self_p->txsize -= sizeof(*self_p->txframe_p);
             self_p->txframe_p++;
         } else {
             thrd_resume_isr(self_p->thrd_p, 0);
+            self_p->thrd_p = NULL;
         }
     }
 
     /* Handle RX frame available interrupt. */
-    if ((interrupt & ESP32_CAN_INT_RX) != 0) {
+    if (interrupt & ESP32_CAN_INT_RX) {
         read_frame_from_hw(self_p, regs_p);
     }
 
     /* Handle error interrupts. */
-    if ((interrupt & (ESP32_CAN_INT_ERR
-                      | ESP32_CAN_INT_DATA_OVERRUN
-                      | ESP32_CAN_INT_WAKEUP
-                      | ESP32_CAN_INT_ERR_PASSIVE
-                      | ESP32_CAN_INT_ARB_LOST
-                      | ESP32_CAN_INT_BUS_ERR)) != 0) {
+    if (interrupt & (ESP32_CAN_INT_ERR
+                     | ESP32_CAN_INT_DATA_OVERRUN
+                     | ESP32_CAN_INT_WAKEUP
+                     | ESP32_CAN_INT_ERR_PASSIVE
+                     | ESP32_CAN_INT_ARB_LOST
+                     | ESP32_CAN_INT_BUS_ERR)) {
         fs_counter_increment(&errors, 1);
+
+        /* In case of many errors or bus-off state reset the hardware */
+        if (regs_p->STATUS & (ESP32_CAN_STATUS_ERR | ESP32_CAN_STATUS_BUS)) {
+
```

Comment:
Sorry, I mean blank line before the ``while``/``if`` row, and after the closing brace of the block following the statement. So no blank line after the opening brace.